openedx/adg/lms/applications/tests/test_helpers.py

```diff
@@ -352,3 +367,170 @@ def test_has_admin_permissions(mocker, is_business_line_admin, is_superuser, is_
+    """
+    Test the course card information when the user has not yet enrolled in a course
+    """
+    user = UserFactory(username=USERNAME, password=PASSWORD)
+    course = CourseOverviewFactory()
+
+    course_cards, is_any_course_started, is_locked = get_course_card_information(user, [course])
+
+    assert course_cards == [{
+        'course': course,
+        'status': 'Not Started',
+        'grade': '',
+        'message': ''
+    }]
+    assert not is_any_course_started
+    assert not is_locked
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'all_modules_attempted, status, grade, message',
+    [
+        (False, 'In Progress', '', ''),
+        (True, 'Re-Take Course', '0', RETAKE_COURSE_MESSAGE)
+    ],
+    ids=['enrolled_in_course', 'failed_in_course']
+)
+def test_get_course_card_information_after_enrolling_and_failing_course(
+    all_modules_attempted, status, grade, message, mocker
+):
+    """
+    Test the course card information when the user has enrolled in a course and when the user has failed the course
+    """
+    mocker.patch('openedx.adg.lms.applications.helpers.has_attempted_all_modules', return_value=all_modules_attempted)
+
+    user = UserFactory(username=USERNAME, password=PASSWORD)
+    course = CourseOverviewFactory()
+    CourseEnrollmentFactory(user=user, course=course)
+
+    course_cards, is_any_course_started, is_locked = get_course_card_information(user, [course])
+
+    assert course_cards == [{
+        'course': course,
+        'status': status,
+        'grade': grade,
+        'message': message
+    }]
+    assert is_any_course_started
+    assert not is_locked
+
+
+@pytest.mark.django_db
+def test_get_course_card_information_after_completing_course(mocker):
+    """
+    Test the course card information when the user has completed a course
+    """
+    mocker.patch.object(helpers.CourseGradeFactory, 'read', return_value=Mock(passed=True, percent=1))
+
+    user = UserFactory(username=USERNAME, password=PASSWORD)
+    course = CourseOverviewFactory()
+    CourseEnrollmentFactory(user=user, course=course)
+
+    course_cards, is_any_course_started, is_locked = get_course_card_information(user, [course])
+
+    assert course_cards == [{
+        'course': course,
+        'status': 'Completed',
+        'grade': '100',
+        'message': ''
+    }]
+    assert is_any_course_started
+    assert not is_locked
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'is_passed, status, message, is_course_locked',
+    [
+        (False, 'Locked', LOCKED_COURSE_MESSAGE, True),
+        (True, 'Not Started', '', False),
+    ],
+    ids=['un_passed_prerequisite', 'passed_prerequisite']
+)
+def test_get_course_card_information_of_a_course_with_prerequisite(
+    is_passed, status, message, is_course_locked, mocker
+):
+    """
+    Test the course card information when a course has a prerequisite has not been passed yet and when a course has a
+    prerequisite has been passed
```

Comment:
*Test the course card information when a course has a prerequisite that has not been passed yet and when a course has a prerequisite that has been passed